spec/javascripts/admin/analytics-modules/ga4-finder-setup.spec.js

```diff
@@ -0,0 +1,251 @@
+            <input class="gem-c-input govuk-input govuk-input--width-4" name="year" type="text">
+        </div>
+    </div>
+  `
+
+  const checkboxes = document.createElement('div')
+  checkboxes.innerHTML = `
+    <fieldset class="govuk-fieldset" aria-describedby="checkboxes-4d23386d-hint">
+      <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">What is your favourite colour?</legend>
+      <div id="checkboxes-4d23386d-hint" class="govuk-hint">Select all that apply</div>
+      <div class="govuk-checkboxes">
+          <div class="govuk-checkboxes__item">
+              <input type="checkbox" name="favourite_colour[]" id="checkboxes-4d23386d-0" value="red" class="govuk-checkboxes__input">
+              <label for="checkboxes-4d23386d-0" class="govuk-label govuk-checkboxes__label">Red</label>
+          </div>
+      </div>
+    </fieldset>
+  `
+
+  const radios = document.createElement('div')
+  radios.innerHTML = `
+    <fieldset class="govuk-fieldset">
+      <div class="govuk-radios">
+          <input type="radio" name="radio-group" id="radio-77f4e0a5-0" value="government-gateway" class="govuk-radios__input">
+          <label for="radio-77f4e0a5-0" class="gem-c-label govuk-label govuk-radios__label">Use Government Gateway</label>
+      </div>
+    </fieldset>
+  `
+
+  const select = document.createElement('div')
+  select.innerHTML = `
+    <select>
+      <option value="1">1</option>
+      <option value="2">2</option>
+    </select>
+  `
+
+  const selectMultiple = document.querySelector('div')
+  selectMultiple.innerHTML = select.innerHTML
+  selectMultiple.querySelector('select').setAttribute('multiple', true)
+
+  const oneDate = document.createElement('div')
+  oneDate.innerHTML = `
+    <input type="text" data-ga4-change-category="update-filter one-date" />
+  `
+
+  const clearAllLink = document.createElement('div')
+  clearAllLink.setAttribute('data-ga4-filter-type', 'filter-type')
+  clearAllLink.innerHTML = `
+    <a href='/' data-ga4-change-category="clear-all-filters">
+      Link Text
+    </a>
+  `
+
+  const trackedInputs = [
+    date,
+    radios,
+    checkboxes,
+    select,
+    selectMultiple,
+    oneDate
+  ]
+
+  container.appendChild(form)
+  document.body.appendChild(container)
+
+  beforeEach(() => {
+    form.innerHTML = ''
+  })
+
+  describe('on change within tracked form', () => {
+    let trackChangeEvent
+
+    const addInputToFormAndChange = (input) => {
+      trackChangeEvent.calls.reset()
+
+      form.innerHTML = input.outerHTML
+      const field = form.querySelector('input, select')
+
+      GOVUK.analyticsGa4.analyticsModules.Ga4FinderSetup.init()
+
+      if (field.tagName === 'SELECT') {
+        field.querySelectorAll('option')[1].selected = true
+        field.dispatchEvent(new Event('change', { bubbles: true }))
+      } else {
+        field.click()
+        field.value = '123'
+        field.dispatchEvent(new Event('change', { bubbles: true }))
+      }
+    }
```

Comment:
I'd move this inside the ` it('calls `trackChangeEvent` on tracked inputs', ` block, as it's only being used there. Better to keep it all together so that it's clear there aren't multiple tests relying on this setup code.